lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -1324,18 +1324,18 @@ tokenize_alias(Rest, Line, Column, Unencoded, Atom, Length, Ascii, Special, Scop
       error(Reason, Unencoded ++ Rest, Scope, Tokens);
 
     true ->
-      AliasesToken = {alias, {Line, Column, Unencoded}, Atom},
+      AliasesToken = {alias, {Line, Column, {Unencoded, nil}}, Atom},
       tokenize(Rest, Line, Column + Length, Scope, [AliasesToken | Tokens])
   end.
 
 %% Check if it is a call identifier (paren | bracket | do)
 
-check_call_identifier(Line, Column, Unencoded, Atom, [$( | _]) ->
-  {paren_identifier, {Line, Column, Unencoded}, Atom};
-check_call_identifier(Line, Column, Unencoded, Atom, [$[ | _]) ->
-  {bracket_identifier, {Line, Column, Unencoded}, Atom};
-check_call_identifier(Line, Column, Unencoded, Atom, _Rest) ->
-  {identifier, {Line, Column, Unencoded}, Atom}.
+check_call_identifier(Line, Column, Unencoded, Delimiter, Atom, [$( | _]) ->
+  {paren_identifier, {Line, Column, {Unencoded, Delimiter}}, Atom};
```

Comment:
@josevalim I feel using a tuple here is a bit awkward, is it fine if we make this a map?

Reply:
@jonatanklosko my suggestion would be to only included the "Unencoded" version if there is a delimiter. Otherwise, the Unencoded is the same. Theoretically, this means we don't know what is the type of delimiter, but our goal is to deprecate `Foo.'bar'`, so we don't need to support it.

Reply:
In other words, the third element is either nil or Unencoded. The Unencoded tells you if a delimiter was used and its Unencoded value.

Reply:
> my suggestion would be to only included the "Unencoded" version if there is a delimiter

The unencoded version is used in warnings and I think the main difference is that it is always unescaped.

> Theoretically, this means we don't know what is the type of delimiter, but our goal is to deprecate Foo.'bar', so we don't need to support it.

For quoted atoms we always say `"`, so I did the same for consistency, but if that's something we should align, let me know.

Reply:
We could check if there us an Unencoded version and if not, we fallback to the regular atom. Without quotes, they should always match.